src/discussions/posts/post/Post.jsx

```diff
@@ -1,76 +1,164 @@
-                    : intl.formatMessage(messages.posted_on, { time: post.created_at })
-                }
-              </div>
-            </div>
-          </div>
-          <div className="status-icons">
-            { post.abuse_flagged && <FontAwesomeIcon icon={faFlag} /> }
-            { post.pinned && <FontAwesomeIcon icon={faThumbtack} /> }
-          </div>
-        </div>
-        <div className="d-flex">
-          { post.raw_body }
+    <div className="m-1">
+      {props.type === 'question' && <FontAwesomeIcon icon={faQuestionCircle} size="lg" />}
+      {props.type === 'discussion' && <FontAwesomeIcon icon={faComments} size="lg" />}
+      {props.pinned && (
+        <FontAwesomeIcon
+          icon={faThumbtack}
+          size="sm"
+          className="position-relative bg-white rounded"
+          style={{
+            left: '-0.25rem',
+            bottom: '-0.75rem',
+            padding: '0.125rem',
+          }}
+        />
+      )}
+    </div>
+  );
+}
+
+PostTypeIcon.propTypes = {
+  type: PropTypes.string.isRequired,
+  pinned: PropTypes.bool.isRequired,
+};
+
+function PostHeader({
+  type,
+  pinned,
+  title,
+  author,
+  authorImg,
+  updateMsg,
+  updateTimestamp,
+  commentCount,
+}) {
+  return (
+    <div className="d-flex flex-fill">
+      <Avatar className="my-1" alt={author} src={authorImg} />
+      <div className="d-flex flex-row flex-fill">
+        <PostTypeIcon type={type} pinned={pinned} />
+        <div className="d-flex flex-column flex-fill">
+          <span className="d-flex font-weight-bold text-gray">
+            {title}
+          </span>
+          <span className="d-flex small text-gray-300">
+            {author}{' | '}
+            <span title={updateTimestamp}>
+              {updateMsg}
+            </span>
+          </span>
         </div>
-        <div className="d-flex">
-          { post.following
-            ? <FontAwesomeIcon icon={faSolidStar} />
-            : <FontAwesomeIcon icon={faEmptyStar} /> }
-          <span className="badge">
-            <FontAwesomeIcon icon={faComments} /> { post.comment_count }
+        <div className="d-flex align-items-center mr-3">
+          <FontAwesomeIcon icon={faComments} className="mr-2" />
+          <span style={{ minWidth: '2rem' }}>
+            {commentCount}
           </span>
         </div>
       </div>
     </div>
   );
 }
 
+PostHeader.propTypes = {
+  type: PropTypes.string.isRequired,
+  pinned: PropTypes.bool.isRequired,
+  title: PropTypes.string.isRequired,
+  author: PropTypes.string.isRequired,
+  authorImg: PropTypes.string.isRequired,
+  updateMsg: PropTypes.string.isRequired,
+  updateTimestamp: PropTypes.string.isRequired,
+  commentCount: PropTypes.number.isRequired,
+};
+
+function Post({
+  post,
+  intl,
+}) {
+  const dispatch = useDispatch();
+
+  return (
+    <div className="d-flex flex-column mt-2">
+      <PostHeader
```

Comment:
Nit: Just passing in a single `post` prop makes this much smaller and doesn't really impact readability.